x-pack/plugins/lens/public/indexpattern_plugin/to_expression.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+
+import { IndexPatternPrivateState } from './indexpattern';
+
+export function toExpression(state: IndexPatternPrivateState) {
+  if (state.columnOrder.length === 0) {
+    return '';
+  }
+
+  const fieldNames = state.columnOrder.map(col => state.columns[col].sourceField);
+  const sortedColumns = state.columnOrder.map(col => state.columns[col]);
+
+  const indexName = state.indexPatterns[state.currentIndexPatternId].title;
+
+  if (sortedColumns.every(({ operationType }) => operationType === 'value')) {
+    return `esdocs index="${indexName}" fields="${fieldNames.join(', ')}" sort="${
+      fieldNames[0]
+    }, DESC"`;
+  } else if (sortedColumns.length) {
+    const aggs = sortedColumns
```

Comment:
It looks like esaggs wants to have the metric columns first in the `aggs` parameter. So for mapping we have to change the order to first metric columns, then the bucket columns from most outer bucket to inner bucket.
Seems like this has something to do with esaggs mapping directly to the UI of the old editor.

It doesn't work correctly if we don't do this:
```
esaggs index="d3d7af60-4c81-11e8-b3d7-01146121b73d" metricsAtAllLevels="false" partialRows="false" 
  aggConfigs="[{\"id\":\"54d00fc9-2020-47cb-a24b-8172abe3d62e\",\"enabled\":true,\"type\":\"date_histogram\",\"schema\":\"segment\",\"params\":{\"field\":\"timestamp\",\"timeRange\":{\"from\":\"now-1d\",\"to\":\"now\"},\"useNormalizedEsInterval\":true,\"interval\":\"1h\",\"drop_partials\":false,\"min_doc_count\":1,\"extended_bounds\":{}}},{\"id\":\"0a082a80-c422-47db-813b-454c3d492277\",\"enabled\":true,\"type\":\"count\",\"schema\":\"metric\",\"params\":{}}]"
``` (first date_histogram, then count)

results in a data table with only the date histogram. If you configure the same thing in the old editor, the aggs are switched around. Multiple bucketing aggs have their "real" order, though

Reply:
I looked into this a bit more. The issue here is actually that the arguments to the Terms aggregation specify `orderBy: '1'` which is not a valid thing to order by. So I need to update that clause of the Terms aggregation.